bindings/go/src/fdb/fdb.go

```diff
@@ -195,48 +194,74 @@ func MustGetAPIVersion() int {
 var apiVersion int
 var networkStarted bool
 var networkMutex sync.RWMutex
+var networkRunning sync.WaitGroup
 
 var openDatabases sync.Map
 
-func startNetwork() error {
+// ensureNetworkIsStarted ensures that the API version is set and then starts the internal network event loop, if not already done.
+// The provided function is guaranteed to be run while network thread is running.
+func ensureNetworkIsStarted(f func()) error {
```

Comment:
```suggestion
func executeWithRunningNetworkThread(f func()) error {
```

I don't want to be nitpicking about the method name but `executeWithRunningNetworkThread` might be a better name for what this method is doing now.

Should we add a check in this method if the network was stopped? And if so return an according error? Right now we would still execute `f()` which would probably result in a panic?

Reply:
Sure; made some changes, please review?

Reply:
Note: I could not use a FoundationDB error code for these, as they do not exist.